### DISCUSSION TITLE
Resolve Assignees from groups

### DISCUSSION THREAD

**Discussion by Fladerant** [renovatebot/renovate#23735] [Suggest an Idea]
### Tell us more.

# Introduction
Hello,
we are having problems with our on prem GitLab and the feature `RENOVATE_ASSIGNEES_FROM_CODE_OWNERS`. Our expected behaviour would be that all users of a group listed in a CODEOWNERS file get added as Assignees to a Merge Request, but it doesn't work and throws an "addAssignees Error". The Problem is the missing implementation of the group-lookup in `addAssignees()` if no corresponding user is found. This was already implemented in #19469 for reviewers with `getMemberUserIDs()` in `addReviewers()` , which resolves the users of a group found in a CODEOWNERS file which then can be added as Reviewers to a Merge Request.

# References
File: renovate/lib/modules/platform/gitlab/index.ts
### Working implementation for Reviewers in addReviewers(): Line 1077

```javascript
let newReviewerIDs: number[];
  try {
    newReviewerIDs = (
      await p.all(
        newReviewers.map((r) => async () => {
          try {
            return [await getUserID(r)];
          } catch (err) {
            // Unable to fetch userId, try resolve as a group
            return getMemberUserIDs(r);
          }
        })
      )
    ).flat();
  } catch (err) {
    logger.warn({ err }, 'Failed to get IDs of the new reviewers');
    return;
```

### Missing implementation for Assignees in addAssignees(): Line 1055

```javascript
assignees: string[]
): Promise<void> {
  try {
    logger.debug(`Adding assignees '${assignees.join(', ')}' to #${iid}`);
    const assigneeIds: number[] = [];
    for (const assignee of assignees) {
      assigneeIds.push(await getUserID(assignee));
    }
    const url = `projects/${
      config.repository
    }/merge_requests/${iid}?${getQueryString({
      'assignee_ids[]': assigneeIds,
    })}`;
    await gitlabApi.putJson(url);
  } catch (err) {
    logger.debug({ err }, 'addAssignees error');
    logger.warn({ iid, assignees }, 'Failed to add assignees');
  }
```

**Comment by viceice** [comment]
feel free to open a PR with required changes. 🤗
